src/Compute/Compute/AvailabilitySets/NewAzureAvailabilitySetCommand.cs

```diff
@@ -98,7 +103,7 @@ public override void ExecuteCmdlet()
                     Tags = Tag == null ? null : Tag.Cast<DictionaryEntry>().ToDictionary(d => (string)d.Key, d => (string)d.Value)
                 };
 
-                if (!string.IsNullOrEmpty(this.Sku))
+                if (this.MyInvocation.BoundParameters.ContainsKey("Sku"))
```

Comment:
Instead of this pattern, you can use the IsParameterBound extension method, which allows you to select by property name, rather than use strings, and makes for cleaner code.